tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Printers.scala

```diff
@@ -80,180 +89,253 @@ object Printers {
     def printTree(tree: Tree, isStat: Boolean): Unit = {
       tree match {
         case EmptyTree =>
-          print("<empty>")
+          printString("<empty>")
 
         // Comments
 
-        case DocComment(text) =>
-          val lines = text.split("\n").toList
+        case tree: DocComment =>
+          val lines = tree.text.split("\n").toList
           if (lines.tail.isEmpty) {
-            print("/** ", lines.head, " */")
+            printString("/** ")
+            printOne(lines.head)
+            printString(" */")
           } else {
-            print("/** ", lines.head); println()
-            for (line <- lines.tail) {
-              print(" *  ", line); println()
+            printString("/** ")
+            printOne(lines.head)
+            println()
+            var rest = lines.tail
+            while (rest.nonEmpty) {
+              printString(" *  ")
+              printOne(rest.head)
+              println()
+              rest = rest.tail
             }
-            print(" */")
+            printString(" */")
           }
 
         // Definitions
 
-        case VarDef(ident, rhs) =>
-          print("var ", ident)
-          if (rhs != EmptyTree)
-            print(" = ", rhs)
+        case tree: VarDef =>
+          printString("var ")
+          printOne(tree.name)
+          if (tree.rhs != EmptyTree) {
+            printString(" = ")
+            printOne(tree.rhs)
+          }
 
-        case Let(ident, mutable, rhs) =>
-          print(if (mutable) "let " else "const ", ident)
-          if (rhs != EmptyTree)
-            print(" = ", rhs)
+        case tree: Let =>
+          printString(if (tree.mutable) "let " else "const ")
+          printIdent(tree.name)
+          if (tree.rhs != EmptyTree) {
+            printString(" = ")
+            printOne(tree.rhs)
+          }
 
-        case ParamDef(ident, rest) =>
-          if (rest)
-            print("...")
-          print(ident)
+        case tree: ParamDef =>
+          if (tree.rest)
+            printString("...")
+          printOne(tree.name)
 
         // Control flow constructs
 
-        case Skip() =>
-          print("/*<skip>*/")
+        case _: Skip =>
+          printString("/*<skip>*/")
 
-        case tree @ Block(trees) =>
+        case tree: Block =>
```

Comment:
Note that `Block` is an exception to my earlier comment because it is *not* a case class. So here it should improve things to have `tree: Block` indeed.